olive/engine/engine.py

```diff
@@ -588,6 +588,8 @@ def run_search(
             time_diff = time.time() - start_time
             self.search_strategy.check_exit_criteria(iter_num, time_diff, signal)
 
+        self.footprints[accelerator_spec].to_file(output_dir / f"{prefix_output_name}footprints.json")
```

Comment:
I noticed this was recently removed in favor of only returning the pareto footprints, but I'd like to keep them somehow. It's a useful blueprint of all the steps that were processed; not just the pareto frontier step. For example, it allows the caller to query the location of the unoptimized models to do something else with them.

Reply:
This sounds good to me. It's definitely useful to have the full footprint saved as well.